bitnami/contour/templates/envoy/deployment.yaml

```
@@ -0,0 +1,267 @@
+{{- if .Values.envoy.enabled }}
+{{- if eq .Values.envoy.kind "Deployment" }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "common.names.fullname" . }}-envoy
```

Let's ensure we comply with the max-length allowed for DNS names in K8s:
```suggestion
  name: {{ printf "%s-envoy" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" }}
```